waddrmgr/scoped_manager.go

```diff
@@ -308,69 +308,92 @@ func (s *ScopedKeyManager) loadAccountInfo(ns walletdb.ReadBucket,
 		return nil, maybeConvertDbError(err)
 	}
 
-	// Ensure the account type is a default account.
-	row, ok := rowInterface.(*dbDefaultAccountRow)
-	if !ok {
-		str := fmt.Sprintf("unsupported account type %T", row)
-		return nil, managerError(ErrDatabase, str, nil)
-	}
+	decryptKey := func(cryptoKey EncryptorDecryptor,
+		encryptedKey []byte) (*hdkeychain.ExtendedKey, error) {
 
-	// Use the crypto public key to decrypt the account public extended
-	// key.
-	serializedKeyPub, err := s.rootManager.cryptoKeyPub.Decrypt(row.pubKeyEncrypted)
-	if err != nil {
-		str := fmt.Sprintf("failed to decrypt public key for account %d",
-			account)
-		return nil, managerError(ErrCrypto, str, err)
-	}
-	acctKeyPub, err := hdkeychain.NewKeyFromString(string(serializedKeyPub))
-	if err != nil {
-		str := fmt.Sprintf("failed to create extended public key for "+
-			"account %d", account)
-		return nil, managerError(ErrKeyChain, str, err)
+		serializedKey, err := cryptoKey.Decrypt(encryptedKey)
+		if err != nil {
+			return nil, err
+		}
+		return hdkeychain.NewKeyFromString(string(serializedKey))
 	}
 
-	// Create the new account info with the known information.  The rest of
+	// Create the new account info with the known information. The rest of
 	// the fields are filled out below.
-	acctInfo := &accountInfo{
-		acctName:          row.name,
-		acctKeyEncrypted:  row.privKeyEncrypted,
-		acctKeyPub:        acctKeyPub,
-		nextExternalIndex: row.nextExternalIndex,
-		nextInternalIndex: row.nextInternalIndex,
-	}
+	var acctInfo *accountInfo
+	watchOnly := s.rootManager.WatchOnly()
+	private := !s.rootManager.isLocked()
+	switch row := rowInterface.(type) {
```

Comment:
So if some wallet actually started using the watch only accounts this still shouldn't break because they'd still be stored as the old, default account type? And new watch-only accounts would be stored under the correct, new type?

Reply:
> So if some wallet actually started using the watch only accounts this still shouldn't break because they'd still be stored as the old, default account type? And new watch-only accounts would be stored under the correct, new type?

Correct. I haven't actually confirmed this, but as @Roasbeef mentioned, this would mean someone interacted with the wallet without going through lnd.

Reply:
Yes, lnd users wouldn't be affected. I meant other wallets that use this code base (if there are any?), wouldn't we change the behavior for them, possibly?